__init__.py

```diff
@@ -36,27 +36,93 @@ def initialize(self):
         self.install_word, self.remove_word = self.translate_list('action')
         self.yes_words = set(self.translate_list('yes'))
 
+    def load_skills_data(self):
+        """Compatibility wrapper. Remove in 18.08"""
+        return getattr(SkillManager, 'load_skills_data', lambda: {})()
+
+    def write_skills_data(self, skills_data):
+        """Compatibility wrapper. Remove in 18.08"""
+        getattr(SkillManager, 'write_skills_data', lambda x: None)(skills_data)
+
     @intent_file_handler('install.intent')
     def install(self, message):
-        def check_installed(skill):
-            if skill.is_local:
+        with self.handle_msm_errors(message.data['skill'], self.remove_word):
+            skill = self.find_skill(message.data['skill'], False)
+            skills_data = self.load_skills_data()
+            skill_data = skills_data.setdefault(skill.name, {})
+            skill_data['manual_install'] = True
+            was_beta = skill_data.get('beta')
+            skill_data['beta'] = False
+
+            if not was_beta and skill.is_local:
                 raise AlreadyInstalled(skill.name)
 
-        self.confirmable_interaction(
-            SkillEntry.install, False, check_installed, message.data,
-            self.install_word, 'install.confirm', 'install.complete'
-        )
+            self.write_skills_data(skills_data)
```

Comment:
Should the beta value be set to false and skill data be written before the action is confirmed?

I believe I got it to switch the beta flag to false and keep the beta version by not replying to the confirmation

Reply:
Yep! Nice catch!